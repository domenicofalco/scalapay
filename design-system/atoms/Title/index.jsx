import PropTypes from "prop-types";
import styled from "styled-components";
import { base, medium, large, xLarge, black } from "fonts";
import { grey900 } from "colors";

const commonStyles = {
  color: grey900,
  fontWeight: black
};

const H1 = styled.h1`
  font-size: ${xLarge};
  ${commonStyles}
`;

const H2 = styled.h2`
  font-size: ${large};
  ${commonStyles}
`;

const H3 = styled.h3`
  font-size: ${medium};
  ${commonStyles}
`;

const H4 = styled.h4`
  font-size: ${base};
  ${commonStyles}
`;

const Title = ({ children, variant, ...rest }) => {
  const El = Title.variants[variant] || primary;
  return (
    <El {...rest} data-variant={variant}>
      {children}
    </El>
  );
};

Title.variants = {
  primary: H1,
  secondary: H2,
  tertiary: H3,
  quaternary: H4
};

Title.defaultProps = {
  variant: "primary"
};

Title.propTypes = {
  children: PropTypes.string.isRequired,
  variant: PropTypes.string.isRequired
};

export default Title;
