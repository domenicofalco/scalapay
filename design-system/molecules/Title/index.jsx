import PropTypes from "prop-types";
import styled from "styled-components";
import { medium, large, xLarge, black } from "fonts";
import { grey900 } from "colors";

const commonStyles = {
  color: grey900,
  fontWeight: black
};

const H1 = styled.h1`
  font-size: ${xLarge};
  ${commonStyles}
`;

const H2 = styled.h2`
  font-size: ${large};
  ${commonStyles}
`;

const H3 = styled.h3`
  font-size: ${medium};
  ${commonStyles}
`;

const Title = ({ children, variant, ...rest }) => {
  const El = Title.variants[variant];
  return <El {...rest}>{children}</El>;
};

Title.variants = {
  primary: H1,
  secondary: H2,
  tertiary: H3
};

Title.defaultProps = {
  variant: "primary"
};

Title.propTypes = {
  children: PropTypes.string.isRequired,
  variant: PropTypes.string.isRequired
};

export default Title;
