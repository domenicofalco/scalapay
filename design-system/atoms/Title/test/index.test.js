import React from "react";
import { cleanup, render } from "@testing-library/react";
import Title from "../";

afterEach(cleanup);

describe("atoms => Title", () => {
  it("It should render 'primary' variant as default prop is missing or wrong", () => {
    const { getByText } = render(<Title type="button">Title</Title>);
    const component = getByText(/Title/i);
    expect(component.dataset.variant).toBe("primary");
  });

  it("It should render 'primary' variant", () => {
    const { getByText, asFragment } = render(
      <Title variant="primary" type="button">
        Title
      </Title>
    );
    const component = getByText(/Title/i);
    expect(component.dataset.variant).toBe("primary");
    expect(asFragment()).toMatchSnapshot();
  });

  it("It should render 'secondary' variant", () => {
    const { getByText } = render(
      <Title variant="secondary" type="button">
        Title
      </Title>
    );
    const component = getByText(/Title/i);
    expect(component.dataset.variant).toBe("secondary");
  });

  it("It should render 'tertiary' variant", () => {
    const { getByText } = render(
      <Title variant="tertiary" type="button">
        Title
      </Title>
    );
    const component = getByText(/Title/i);
    expect(component.dataset.variant).toBe("tertiary");
  });

  it("It should render 'quaternary' variant", () => {
    const { getByText } = render(
      <Title variant="quaternary" type="button">
        Title
      </Title>
    );
    const component = getByText(/Title/i);
    expect(component.dataset.variant).toBe("quaternary");
  });
});
