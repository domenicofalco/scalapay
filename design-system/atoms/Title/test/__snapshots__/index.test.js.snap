// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms => Title It should render 'primary' variant 1`] = `
<DocumentFragment>
  <h1
    class="Title__H1-sc-1jtvfjh-0 DJZNc"
    data-variant="primary"
    type="button"
  >
    Title
  </h1>
</DocumentFragment>
`;
